spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -569,6 +566,15 @@ public final void initialize(String name, CaseInsensitiveStringMap options) {
       }
     }
 
+    if (sparkSession
```

Comment:
What about adding a helper method in the utility to validate this? Then we can avoid complicating this class and can reuse that method here and in the read config object.

```
SparkUtil.validateNoLegacyTimestampWithoutZoneConfigs(sparkSession);
```

The name above is random.

Reply:
Thanks, I've moved the checks to `SparkUtil` 👍🏻